In [1]:
!pip install numpy

In [17]:
# Problem 1: Hypothesis Testing
# 1. 
import numpy as np
myFile = open('engagement_1.txt')
data = myFile.readlines()
myFile.close()
data = [float(x) for x in data]
avg = np.mean(data)
print("The mean of the data is " + str(avg))


The mean of the data is 0.7430304110448239


1. 
H0: u = .75 The average engagement of students who became knowledgeable in the material is .75 (null)
H1: u!= .75 The average engagement of students who became knowledgeable in the material is not .75 (alt)

We can use a hypothesis test to determine whether or not our null or alternative hypotheses are true.

In [44]:
# Problem 1: Hypothesis Testing
# 2.   
u = .75
# Report the sample size, 
# the sample mean, 
# the standard error, 
# the standard score, 
# and the p-value.
import numpy as np
from scipy.stats import norm
myFile = open('engagement_1.txt')
data = myFile.readlines()
myFile.close()
data = [float(x) for x in data]
# finding sample size
sampleSize = len(data)
print("The sample size of the data is " + str(sampleSize))
# finding mean
mean_of_data = np.mean(data)
print("The mean of the data is " + str(mean_of_data))
# finding SD
sd = np.std(data, ddof=1)
#print("The standard deviation of the data is " + str(sd))
# finding the standard error
SE = sd / np.sqrt(sampleSize)
print("The standard error of the data is " + str(SE))
# finding the z score (standard score)
ss = (mean_of_data - u)/SE
print("The standard score of the data is " + str(ss))
#p = norm.cdf((ss))
p = 2 * norm.cdf(-abs(ss))
print("The p value of the data is " + str(p))

The sample size of the data is 937
The mean of the data is 0.7430304110448239
The standard error of the data is 0.004153027288269652
The standard score of the data is -1.6781948375012814
The p value of the data is 0.09330906925243751


#2. reject .1 null
    fail to reject .05
    fail to reject .01
    At  a confidence level of 0.1 the result is significant, but at a confidence level of 0.01 and 0.05 it is not. We can reject the null hypothesis at an alpha level of 0.1 but we fail to reject it at 0.05 and 0.01.

In [28]:
# Problem 1: Hypothesis Testing
# 3.  
p = 0.05 #confidence level
# find the largest standard error at this level
# find the minimum sample size
z_c = norm.ppf(p)
# use z score to find standard error
minSE = (mean_of_data - u)/z_c
print("The highest possible standard error with a confidence level of 0.05 is " + str(minSE))
minSampleSize = (sd/minSE)**2
print("The lowest possible sample size with a confidence level of 0.05 is " + str(minSampleSize))


The highest possible standard error with a confidence level of 0.05 is 0.004237209220916103
The lowest possible sample size with a confidence level of 0.05 is 900.1385114802096


#3. The largest standard error possible at a confidence level of 0.05 is .0042372 and the smallest possible sample size is 900.


In [ ]:
# Problem 1: Hypothesis Testing
# 4. 

#4.
H0: u0 = u1 
The average engagement of students who became knowledgeable in the material is different between engagement 1 and those who did not in engagement 2 (null)
H1: u0!= u1
The average engagement of students who became knowledgeable in the material is not different between engagement 1 and and those who did not in engagement 2 (alt)
We can use a two sample z test to validate this belief.


In [8]:
# Problem 1: Hypothesis Testing
# 5. 
import numpy as np
from scipy.stats import norm
from scipy.stats import t
u = 0
myFile = open('engagement_1.txt')
data = myFile.readlines()
myFile.close()
engagement1 = [float(x) for x in data]
engagement1Length = len(engagement1)
print("The sample size of engagement 1 is " + str(engagement1Length))
engagement1Mean = np.mean(engagement1)
print("The mean of engagement 1 is " + str(engagement1Mean))
engagement1SE = (np.std(engagement1, ddof=1)**2) / engagement1Length
print("The standard error of engagement 1 is " + str(engagement1SE))


myFile = open('engagement_0.txt')
data = myFile.readlines()
myFile.close()
engagement0 = [float(x) for x in data]
engagement0Length = len(engagement0)
print("The sample size of engagement 0 is " + str(engagement0Length))
engagement0Mean = np.mean(engagement0)
print("The mean of engagement 0 is " + str(engagement0Mean))
engagement0SE = (np.std(engagement0, ddof=1)**2) / engagement0Length
print("The standard error of engagement 0 is " + str(engagement0SE))

zScore = (engagement0Mean - engagement1Mean)/ (np.sqrt(engagement0SE + engagement1SE))
print("The z score of the two populations is " + str(zScore))

#p = t.cdf(abs(zScore), (engagement0Length + engagement1Length))
p = 2 * norm.cdf(-abs(zScore))
print("The p value of the two populations is " + str(p))


The sample size of engagement 1 is 937
The mean of engagement 1 is 0.7430304110448239
The standard error of engagement 1 is 1.7247635657112375e-05
The sample size of engagement 0 is 1977
The mean of engagement 0 is 0.6399545077035914
The standard error of engagement 0 is 3.2672536978964496e-05
The z score of the two populations is -14.588784540028351
The p value of the two populations is 3.3104307168195455e-48


#5.
The sample size of engagement 1 is 937
The mean of engagement 1 is 0.7430304110448239
The standard error of engagement 1 is 1.7247635657112375e-05
The sample size of engagement 0 is 1977
The mean of engagement 0 is 0.6399545077035914
The standard error of engagement 0 is 3.2672536978964496e-05
The z score of the two populations is -14.588784540028351
The p value of the two populations is 3.3104307168195455e-48
Under the 0.1, 0.05, and 0.01 confidence levels, we can reject our null hypothesis of the average engagement of students who became knowledgeable in the material is different between engagement 1 and those who did not in engagement 2